src/platform-aws.c

```diff
@@ -97,207 +102,168 @@ static const char* get_platform_type(void)
-			if (rc) {
-				NCCL_OFI_WARN("Error setting NCCL_PROTO environment variable: %d", rc);
-				ret = ncclSystemError;
-				goto exit;
-			}
-		} else if (strcmp(getenv("NCCL_PROTO"), "simple")) {
-			NCCL_OFI_WARN("NCCL_PROTO was set to \"LL/LL128\", but the Libfabric endpoint does not support 128 byte in-order aligned stores. This endpoint may corrupt data during communication");
-		}
+	for (size_t idx = 0; idx < platform_n; idx++) {
+		if (strcmp(platform_type, platform_data_map[idx].name) == 0)
+			platform_data = &platform_data_map[idx];
 	}
 
-exit:
-	return ret;
+	pthread_mutex_unlock(&mutex);
+
+	return platform_data;
 }
 
-static ncclResult_t validate_rdma_write(struct fid_ep *ep)
+/*
+ * validate that EFA is using RDMA write natively and not in an
+ * emulated fasion.
+ */
+static int validate_rdma_write(struct fid_ep *ep)
 {
-	int ret = ncclSuccess;
+	int ret = 0;
 #if HAVE_DECL_FI_OPT_EFA_EMULATED_WRITE
 	bool optval;
 	size_t optlen = sizeof(optval);
 
 	ret = fi_getopt(&ep->fid, FI_OPT_ENDPOINT, FI_OPT_EFA_EMULATED_WRITE, &optval, &optlen);
-	if(ret != 0 || optlen != sizeof(optval)) {
-		NCCL_OFI_WARN("Couldn't get FI_OPT_EFA_EMULATED_WRITE. optlen: %lu, RC: %d, ERROR: %s",
-			      optlen, ret, fi_strerror(-ret));
-		ret = ncclSystemError;
+	if (ret != 0) {
+		NCCL_OFI_WARN("Couldn't get FI_OPT_EFA_EMULATED_WRITE. RC: %d, ERROR: %s",
+			      ret, fi_strerror(-ret));
 		goto exit;
-	}
-	/* If the selected protocol is RDMA write and RDMA write is not
-	 * supported for the endpoint, throw an error 
-	 */
-	else if (optval && 0 == strcmp("RDMA", nccl_ofi_selected_protocol)) {
+	} else if (optlen != sizeof(optval)) {
+		NCCL_OFI_WARN("Unexpected response size when checking FI_OPT_EFA_EMULATED_WRITE.  Expected %lu, got %lu",
+			      sizeof(optval), optlen);
+		ret = -EINVAL;
+		goto exit;
+	} else if (optval) {
 		NCCL_OFI_WARN("FI_OPT_EFA_EMULATED_WRITE is true when the communication protocol is RDMA write.");
-		ret = ncclSystemError;
+		ret = -EINVAL;
 		goto exit;
 	}
 	NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET, "Get endpoint option FI_OPT_EFA_EMULATED_WRITE. optval: %d", 
 		       optval);
 #else
 	NCCL_OFI_WARN("FI_OPT_EFA_EMULATED_WRITE not declared when the communication protocol is RDMA write.");
-	ret = ncclSystemError;
+	ret = -EINVAL;
 	goto exit;
 #endif
+
 exit:
 	return ret;
 }
 
-static ncclResult_t configure_sendrecv_inorder(struct fid_ep *ep, bool is_init)
-{
-	int ret = ncclSuccess;
-#if HAVE_DECL_FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES
-	bool optval = true;
 
-	ret = fi_setopt(&ep->fid, FI_OPT_ENDPOINT,
-			FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES, 
-			&optval, sizeof(optval));
-	if (ret != 0 && ret != -FI_EOPNOTSUPP) {
-		NCCL_OFI_WARN("Couldn't set FI_OPT_EFA_SENDRECV_IN_ORDER_ALIGNED_128_BYTES. RC: %d, ERROR: %s",
-			      ret, fi_strerror(-ret));
-		ret = ncclSystemError;
-		goto exit;
-	}
+#if HAVE_CUDA
+static int configure_nccl_proto(bool have_ordering)
```

Comment:
`have_ordering` is an unused parameter in this function, can be removed.

Reply:
Doh; meant to remove that.  Fixed.